### ISSUE TITLE
Add optional "inherits" or "substitute" to sounds.xml

### ISSUE THREAD

**Issue by mscuthbert** [w3c/musicxml#364]
Looking at the discussion of `wind.flutes.whistle.tin.c` (etc.) and "generic" versions of sounds that consumers can use in case they have no more specific version of the instrument, it strikes me that there is still something quite missing from the specification:

 If there is no `wind.flutes.whistle.tin.c` sound, a program can substitute `wind.flutes.whistle.tin` and from there `wind.flutes.whistle` or up to `wind.flutes`, or (in the unlikely event there is no flute sound) `wind`.  The dots specify a hierarchy that gives a reasonable inheritance tree.  Similarly with `brass.trumpet.bflat` etc.

However, there are a number of cases where most programs can find a better substitute rather than merely going up the inheritance tree.  For instance:

`brass.sackbutt.tenor` should rightly inherit from `brass.sackbutt` but from there there is a much better substitute than just `brass` -- `brass.trombone` is greatly preferred.

Similarly:

`brass.sousaphone` => `brass.tuba`
`wind.reed.heckelphone` => `wind.reed.oboe.bass`
`wind.reed.contrabassoon` => `wind.reed.bassoon`
`wind.reed.basset-horn` => `wind.reed.clarinet.alto`
`wind.reed.clarinet.contrabass` => `wind.reed.clarinet.bass`
`wind.reed.clarinet.piccolo.aflat` => `wind.reed.clarinet.eflat`
`pitched-percussion.xylorimba` => `pitched-percussion.marimba`
`pluck.sanshin` => `pluck.shamisen`
`pluck.archlute` => `pluck.theorbo` => `pluck.lute`

etc.

I propose that `<sound>` entries in the sounds.xml file gain a new attribute (element?) `inherits="[other-sound-id]"` which specifies a better sound to choose from than moving up the inheritance chain.  So:

`<sound id="brass.sackbutt" inherits="brass.trombone"/>`

would become acceptable.  However, if this were in place, it would not be acceptable to also add to `brass.trombone` an `inherits="brass.sackbutt"` since this might cause an infinite loop for consumers.

A secondary proposal would be to specify multiple instruments which may substitute for this instrument, but which consumers have to guard against circular data structures.  For instance, this might be desired:

```
<sound id="wind.reed.heckelphone" inherits="wind.reed.oboe" substitutes="wind.reed.oboe.bass,wind.reed.english-horn"/>
<sound id="wind.reed.oboe.bass" substitutes="wind.reed.hecelphone,wind.reed.english-horn"/> <!-- no need for inherits wind.reed.oboe: implied -->
```

this might be too complex for now, and maybe we always know which of two alternates is most likely to exist, but it does allow for a much wider range of good substitute sounds without getting into potentially political decisions (is it `pluck.biwa` => `pluck.pipa` or vice-versa?).

thanks!

**Comment by mdgood** [comment]
@mscuthbert This is an excellent idea, but there are intellectual property issues that keep us from doing this in an open standard for many more years yet.

**Comment by samuelbradshaw** [comment]
I'd propose "fallback" as the name of the attribute on `<sound>`.

**Comment by mscuthbert** [comment]
> @mscuthbert This is an excellent idea, but there are intellectual property issues that keep us from doing this in an open standard for many more years yet.

Ah.  I can guess what that means.  Well, I didn't know this, so at least you can use this as evidence that such an invention is plainly obvious to someone with normal art in the field of study.

**Comment by jsawruk** [comment]
@mscuthbert What about allowing multiple `<instrument-sound>` elements in a `<score-instrument>`? This would allow the author to specify alternate sounds, potentially in order of preference?

Maybe `<virtual-instrument>` could also be allowed multiple times?

**Comment by mdgood** [comment]
@jsawruk To me that approach seems more like an application issue than a format issue, although I do appreciate it could go either way. Current application support for the sound functionality seems underutilized. I am hesitant to add more features in 4.0 unless they have come up as pain points in specific applications (e.g. #293).

**Comment by jsawruk** [comment]
@mdgood Agreed, no need to add more features to 4.0, but might be something to consider in future versions.
